code/modules/client/preferences.dm

```diff
@@ -1349,7 +1359,17 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 						real_name = new_name
 					else
 						to_chat(user, "<font color='red'>Invalid name. Your name should be at least 2 and at most [MAX_NAME_LEN] characters long. It may only contain the characters A-Z, a-z, -, ' and .</font>")
-
+				//MONKESTATION EDIT START - FLAVOR TEXT
+				if("flavor_text")
+					var/new_text = input(user, "Please enter new flavor text (appears on examine):", "Character Preference") as text|null
+					if(!new_text)
+						return
+					if(length(new_text) > MAX_FLAVOR_LEN)
+						alert("Your flavor text is too long. It must be no more than [MAX_FLAVOR_LEN] characters long. The current text will be trimmed down to meet the limit.")
+						new_text = copytext(new_text, 1, MAX_FLAVOR_LEN+1)
```

Comment:
Why is it MAX_FLAVOR_LEN+1?
The Bee coders were trying to figure this out as well when they were looking into porting the Donk version.

Reply:
as far as I know, copytext copies from the start (1) up to the max (MAX_FLAVOR_LEN+1), will test and update code if needed

Reply:
Not an issue, as per byond docs:
![image](https://user-images.githubusercontent.com/41448081/148815982-0f2998c8-5ef7-40ff-891a-8fc02d1ba837.png)